src/modules/users/users.service.ts

```diff
@@ -35,4 +44,140 @@ export class UsersService {
       throw new Error('Failed to update user');
     }
   }
+
+  async createAdminUser(createUserDto: CreateUserDto): Promise<User> {
+    try {
+      const candidate = await this.findOne({
+        where: { email: createUserDto.email },
+      });
+
+      if (candidate) {
+        throw new BadRequestException(
+          `User with email:${createUserDto.email} already exists`,
+        );
+      }
+      if (createUserDto.role !== UserRole.ADMIN) {
+        throw new BadRequestException(
+          `Wrong role for creating admin: ${createUserDto.role}`,
+        );
+      }
+      const user = this.userRepository.create(createUserDto);
+
+      return await this.userRepository.save(user);
```

Comment:
why we need both create & save ? when you create you already save. I don't like repository, I prefer queryBuilder. But I briefly check documentation I could not find good explanation